packages/lib/SyncTargetJoplinServerSAML.ts

```diff
@@ -72,6 +72,45 @@ export default class SyncTargetJoplinServerSAML extends SyncTargetJoplinServer {
 		return false;
 	}
 
+	public static override async checkConfig(fileApi: FileApiOptions) {
+		try {
+			// Simulate a login request
+			const result = await fetch(`${fileApi.path()}/api/saml`);
+
+			if (result.status === 200) { // The server successfully responded, SAML is enabled
+				return {
+					ok: true,
+					errorMessage: '',
+				};
+			} else { // SAML is disabled or an error occurred
+				let message = result.statusText;
+
+				// Check if we got an error message
+				if (result.headers.get('Content-Type').includes('application/json')) {
+					try {
+						const json = await result.json();
+
+						if (json.error) {
+							message = json.error;
+						}
+					} catch (e) {
+						message = 'Not a valid JSON response';
```

Comment:
Can I suggest setting message to `statusText` in this case as that would give a bit more information? What we usually do is load the response as text, try to parse it, and it fails send the raw text response since that gives the most details. But statusText maybe is good enough here